From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Robinson <pbrobinson@gmail.com>
Date: Fri, 10 Jul 2020 16:18:59 +0100
Subject: [PATCH] Fixes "acpi: prefer booting with ACPI over DTS" to be RHEL
 only

The ACPI prefernce is RHEL specific, in Fedora we use upstream defaults so
it doesn't randomly change the expectations as to how certain hardware works
that generally isn't classed as "enterprise". So wrap it in the special RHEL
check as it should be for the ARK kernels.

Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
CC: Mark Salter <msalter@redhat.com>
---
 arch/arm64/kernel/acpi.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/kernel/acpi.c b/arch/arm64/kernel/acpi.c
index 2e4ba374ab54..7630e22e1722 100644
--- a/arch/arm64/kernel/acpi.c
+++ b/arch/arm64/kernel/acpi.c
@@ -40,7 +40,11 @@ int acpi_pci_disabled = 1;	/* skip ACPI PCI scan and IRQ initialization */
 EXPORT_SYMBOL(acpi_pci_disabled);

 static bool param_acpi_off __initdata;
+#ifdef CONFIG_RHEL_DIFFERENCES
 static bool param_acpi_on __initdata = true;
+#else
+static bool param_acpi_on __initdata;
+#endif
 static bool param_acpi_force __initdata;

 static int __init parse_acpi(char *arg)
-- 
2.28.0

